src/main/java/org/opensearch/ad/transport/ADStatsNodesTransportAction.java

```diff
@@ -11,92 +11,75 @@
 
 package org.opensearch.ad.transport;
 
+import static org.opensearch.ad.constant.CommonErrorMessages.FAIL_TO_GET_NODE_STATS;
+import static org.opensearch.ad.util.RestHandlerUtils.wrapRestActionListener;
+
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
+import org.opensearch.action.ActionListener;
 import org.opensearch.action.FailedNodeException;
 import org.opensearch.action.support.ActionFilters;
-import org.opensearch.action.support.nodes.TransportNodesAction;
+import org.opensearch.action.support.TransportAction;
 import org.opensearch.ad.stats.ADStats;
 import org.opensearch.ad.stats.InternalStatNames;
 import org.opensearch.ad.task.ADTaskManager;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.inject.Inject;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.monitor.jvm.JvmService;
-import org.opensearch.threadpool.ThreadPool;
-import org.opensearch.transport.TransportService;
+import org.opensearch.sdk.SDKClusterService;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskManager;
+
+import com.google.inject.Inject;
 
 /**
  *  ADStatsNodesTransportAction contains the logic to extract the stats from the nodes
  */
-public class ADStatsNodesTransportAction extends
-    TransportNodesAction<ADStatsRequest, ADStatsNodesResponse, ADStatsNodeRequest, ADStatsNodeResponse> {
-
+public class ADStatsNodesTransportAction extends TransportAction<ADStatsNodeRequest, ADStatsNodesResponse> {
```

Comment:
What are we losing by removing this inheritance vs. replacing it with an SDK equivalent?   This approach will require all the classes which inherit from this same superclass to have to make the same changes, while if you update (or replace) the superclass, then that's a one-time change that all the subclasses can use.

Reply:
I have to dig into this. I will post my findings here.

Reply:
@dbwiddis

When ADStatsNodesTransportAction is used in

1) Plugin Architecture: It extends TransportNodesAction. The transportNodesAction Accepts 4 parameters 
(BaseNodesRequest, NodesResponse, NodeRequest, NodeResponse). It can perform Async operation on multiple nodes. 
P.S.:      NodeRequest extends TransportRequest

2) Extension Architecture: It extends TransportAction. The transport action accepts 2 parameters (NodeRequest and NodesResponse). It can only perform request on only 1 node. Therefore in order to provide the functionality similar to TransportNodesAction I can open up a new issue.

As the TransportNodesAction counterpart for extensions needs more understanding and overall scope is beyond GetDetectorStats API.

What would you suggest? Shall I open a new issue?

Reply:
Yeah, this is probably worth a new issue.  Summing it up:
 - `TransportNodesAction` is on OpenSearch meant for plugins, so we can't/shouldn't change it for extensions, but
 - There are some methods that add some functionality to the inheritance from `TransportAction`, specifically the `newResponse` method and the abstract methods.

I'm not sure we need a `SDKTransportNodesAction` but we should at least look into whether it makes sense.

Reply:
As discussed, I will be opening a new issue for this.

Reply:
Let's change the request to `ADStatsRequest`